collect/collect.go

```diff
@@ -47,6 +47,9 @@ var (
 	// Whether or not to use NTLM authentication
 	UseNtlm bool = false
 
+	// DefaultClient can be used to override the HTTP client that will be used to make requests.
+	DefaultClient *http.Client
```

Comment:
This is kinda wierd to me. Collect already has an odd client with a ghetto timeout thing (that looks buggy to me). Can we just have a central utility that sets http.DefaultClient?

Something like `util.SetupHttpClient(appName, msg, features...)`?